### PR TITLE
[chore] Update the CHANGELOG to add details about the new feature that enables reduce_scatter overlap in backward propagation

### PR DESCRIPTION
**Pull request by tmarkstrum**
## What does this PR do?
Updated the change log for enabling reduce_scatter overlap with the rest operations in backward propagation. 

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,9 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.4.5] - TBD
 
 ### Added
+- FSDP: Added process_group_reduce_scatter parameter to allow users to pass in the process group that is used for reduce scatter operation.
```
Comment (anj-s) [inline CHANGELOG.md:10]: Add the PR number?

Comment (anj-s) [inline CHANGELOG.md:10]: Can you add a sentence about what this enable and how to do a rollback as well?

**Comment by anj-s** [comment]
Can you update the PR title to follow the [guidelines](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md#type) such that it says something like "[chore] Update the CHANGELOG to add details about <PR number>"

File: CHANGELOG.md (hunk 1 above)
Comment (tmarkstrum) [reply CHANGELOG.md]: done

Comment (tmarkstrum) [reply CHANGELOG.md]: done

**Review by anj-s — APPROVED** [review]

**Pull request merged**